### PR TITLE
Support expo 49

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Add support for Expo 49 which is currently in beta and should be released this week.
Also updated and fixed the version of yarn used to ensure consistent builds.

### CODE REVIEW

**Review by kayayarai — APPROVED** [review]

**Review by xgouchet — DISMISSED** [review]

File: src/plugin/withIosSourcemaps/withIosSourcemaps.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import type { ConfigPlugin } from "@expo/config-plugins";
 import { withXcodeProject } from "@expo/config-plugins";
 
 const SOURCEMAP_FILE_COMMAND = "export SOURCEMAP_FILE=./main.jsbundle.map";
-const DATADOG_XCODE_COMMAND = `yarn datadog-ci react-native xcode \`\\\"$NODE_BINARY\\\" --print \\\"require('path').dirname(require.resolve('react-native/package.json')) + '/scripts/react-native-xcode.sh'\\\"\``;
+const DATADOG_XCODE_COMMAND = `../node_modules/.bin/datadog-ci react-native xcode \`\\\"$NODE_BINARY\\\" --print \\\"require('path').dirname(require.resolve('react-native/package.json')) + '/scripts/react-native-xcode.sh'\\\"\``;
```
Comment (fuzzybinary) [inline src/plugin/withIosSourcemaps/withIosSourcemaps.ts:11]: Are we at all worried about the "Unnecessary escape character" warnings here? It's really hard to parse whether the warning is right....

**Review by fuzzybinary — APPROVED** [review]

**Review by eliseo-juan — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]
